### PR TITLE
stream Cantabular CSV data, add extra S3 uploader and implement public/private

### PR DESCRIPTION
**Pull request by DavidSubiros**
### What

- Get `isPublished` from instance information
- Used version of dp-api-clients-go with Cantabular CSV stream [dp-api-clients-go PR](https://github.com/ONSdigital/dp-api-clients-go/pull/154)
- Refactored handler code to use the CSV streaming functionality
- Get file size by doing an S3 `HeadObject` call, as we no longer keep the whole data in memory [dp-s3 PR](https://github.com/ONSdigital/dp-s3/pull/18)
- Added an extra S3 uploader, having one for published files and one for private files
- Use kafka send method instead of marshaling and sending bytes to Output channel, and updated tests accordingly [dp-kafka PR](https://github.com/ONSdigital/dp-kafka/pull/44)

### How to review

- Make sure code changes make sense and we no longer keep too much data in memory
- Make sure unit tests pass
- Make sure component tests pass
- (info only) locally tested that the Cantabular import process is still working, after having added the private bucket to minio: [dp-compose pr](https://github.com/ONSdigital/dp-compose/pull/47)

### Who can review

anyone

### CODE REVIEW

File: handler/handler.go [hunk 1]
```diff
@@ -450,6 +356,7 @@ func generatePrivateURL(downloadServiceURL, instanceID string) string {
 }
 
 // generateS3Filename generates the S3 key (filename including `subpaths` after the bucket) for the provided instanceID
+// TODO filename should be datasets/<dataset_name>_<version>.csv to match CMD naming
```
Comment (franmoore05) [inline handler/handler.go:359]: Should this be created as a ticket to ensure it gets picked up?

Comment (DavidSubiros) [reply handler/handler.go:359]: Maybe yes, I started working on making the changes to consume the new topic and schema generated by dataset api (when Neil finishis his changes) - maybe the file renaming should be part of the same ticket?

Comment (DavidSubiros) [reply handler/handler.go:359]: created the following trello card: https://trello.com/c/lLu8DzMx/5398-cantabular-csv-exporter-should-consume-cantabular-export-start-topic

**Review by franmoore05 — APPROVED** [review]
LGTM

**Pull request merged**